### PR TITLE
Add new endpoints for transferring of domain and addresses

### LINKED ISSUE #24: [TypeScript SDK] Add support for xferdomain and xferaddress
Add support for xferdomain and xferaddress

### PR DESCRIPTION
**Pull request by 0xCasey**

### CODE REVIEW

**Review by adsorptionenthalpy — APPROVED** [review]

**Comment by 0xCasey** [comment]
@andreyvEze Please review

File: tests/testnet.spec.js [hunk 1]
```diff
@@ -99,6 +99,7 @@ describe('Testing generic actions', () => {
   const newFioAddress = generateTestingFioAddress(newFioDomain)
   const privateKeyExample = '5Kbb37EAqQgZ9vWUHoPiC2uXYhyGSFNbL6oiDp24Ea1ADxV1qnu'
   const publicKeyExample = 'FIO5kJKNHwctcfUM5XZyiWSqSTM5HTzznJP9F3ZdbhaQAHEVq575o'
+  const newPublicKey = 'FIO7isxEua78KPVbGzKemH4nj2bWE52gqj8Hkac3tc7jKNvpfWzYS'
```
Comment (andreyvEze) [inline tests/testnet.spec.js:102]: don't know should we leave this key here, maybe we could make it empty and rename it to `pubKeyForTransfer`, or just use `publicKey2 `

File: src/transactions/signed/TransferFioDomain.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        this.technologyProviderId = technologyProviderId
+
+        this.validationData = { fioAddress, tpid: technologyProviderId || null }
+        this.validationRules = validationRules.registerFioDomain
+    }
+
+    public getData(): any {
+        const actor = this.getActor()
+        const data = {
+            fio_address: this.fioAddress,
```
Comment (andreyvEze) [inline src/transactions/signed/TransferFioDomain.ts:28]: should it be `fio_domain`, `fioDomain`?

File: src/transactions/signed/TransferFioDomain.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+export class TransferFioDomain extends SignedTransaction {
+    public ENDPOINT: string = 'chain/transfer_fio_domain'
+    public ACTION: string = 'xferdomain'
+    public ACCOUNT: string = Constants.defaultAccount
+    public fioAddress: string
+    public newOwnerKey: string
+    public maxFee: number
+    public technologyProviderId: string
+
+    constructor(fioAddress: string, newOwnerKey: string, maxFee: number, technologyProviderId: string = '') {
```
Comment (andreyvEze) [inline src/transactions/signed/TransferFioDomain.ts:14]: should it be `fioDomain`?

File: src/FIOSDK.ts [hunk 4]
```diff
@@ -947,6 +1011,20 @@ export class FIOSDK {
...
+      case 'transferFioAddress':
+        return this.transferFioAddress(
+            params.fioAddress,
+            params.newOwnerKey,
+            params.maxFee,
+            params.technologyProviderId,
+        )
+      case 'transferFioDomain':
+        return this.transferFioDomain(
+            params.fioAddress,
```
Comment (andreyvEze) [inline src/FIOSDK.ts:1023]: should it be `fioDomain`?

File: src/FIOSDK.ts [hunk 5]
```diff
@@ -370,6 +372,50 @@ export class FIOSDK {
...
+        this.getTechnologyProviderId(technologyProviderId),
+    )
+    return transferFioDomain.execute(this.privateKey, this.publicKey)
+  }
+
+  /**
+   * Transfers a FIO Address on the FIO blockchain.
+   *
+   * @param fioAddress FIO Address to transfer. The owner will be the public key associated with the FIO SDK instance.
+   * @param maxFee Maximum amount of SUFs the user is willing to pay for fee. Should be preceded by @ [getFee] for correct value.
```
Comment (andreyvEze) [inline src/FIOSDK.ts:401]: could we add `newOwnerKey` param in docs

File: src/FIOSDK.ts [hunk 6]
```diff
@@ -370,6 +372,50 @@ export class FIOSDK {
     return registerFioDomain.execute(this.privateKey, this.publicKey)
   }
 
+  /**
+   * Transfers a FIO Domain on the FIO blockchain.
+   *
+   * @param fioDomain FIO Domain to transfer. The owner will be the public key associated with the FIO SDK instance.
+   * @param maxFee Maximum amount of SUFs the user is willing to pay for fee. Should be preceded by @ [getFee] for correct value.
```
Comment (andreyvEze) [inline src/FIOSDK.ts:379]: could we add `newOwnerKey` param in docs

File: lib/FIOSDK.js [hunk 7]
```diff
@@ -626,6 +665,10 @@ class FIOSDK {
                 return this.renewFioDomain(params.fioDomain, params.maxFee, params.technologyProviderId);
             case 'renewFioAddress':
                 return this.renewFioAddress(params.fioAddress, params.maxFee, params.technologyProviderId);
+            case 'transferFioAddress':
+                return this.transferFioAddress(params.fioAddress, params.maxFee, params.technologyProviderId);
+            case 'transferFioDomain':
+                return this.transferFioDomain(params.fioAddress, params.maxFee, params.technologyProviderId);
```
Comment (andreyvEze) [inline lib/FIOSDK.js:671]: could we rebuild, the `newOwnerKey` is missed

File: lib/FIOSDK.js [hunk 8]
```diff
@@ -626,6 +665,10 @@ class FIOSDK {
                 return this.renewFioDomain(params.fioDomain, params.maxFee, params.technologyProviderId);
             case 'renewFioAddress':
                 return this.renewFioAddress(params.fioAddress, params.maxFee, params.technologyProviderId);
+            case 'transferFioAddress':
+                return this.transferFioAddress(params.fioAddress, params.maxFee, params.technologyProviderId);
```
Comment (andreyvEze) [inline lib/FIOSDK.js:669]: could we rebuild, the `newOwnerKey` is missed

File: lib/FIOSDK.js [hunk 9]
```diff
@@ -271,6 +271,29 @@ class FIOSDK {
...
+     *
+     * @param fioDomain FIO Domain to transfer.
+     * @param maxFee Maximum amount of SUFs the user is willing to pay for fee. Should be preceded by @ [getFee] for correct value.
+     * @param technologyProviderId FIO Address of the wallet which generates this transaction.
+     */
+    transferFioDomain(fioDomain, newOwnerKey, maxFee, technologyProviderId = null) {
+        const transferFioDomain = new SignedTransactions.transferFioDomain(fioDomain, newOwnerKey, maxFee, this.getTechnologyProviderId(technologyProviderId));
+        return transferFioDomain.execute(this.privateKey, this.publicKey);
+    }
+    //NEW
```
Comment (andreyvEze) [inline lib/FIOSDK.js:285]: do we need this comment? how its appear here?

File: tests/testnet.spec.js (hunk 1 above)
Comment (0xCasey) [reply tests/testnet.spec.js:102]: I like this approach. Will make corrections.

**Pull request merged**